src/Kestrel.Core/Internal/Http/MessageBody.cs

```diff
@@ -111,6 +112,27 @@ public virtual Task ConsumeAsync()
 
         public abstract Task StopAsync();
 
+        protected void TryProduceContinue()
+        {
+            if (_send100Continue)
+            {
+                _context.HttpResponseControl.ProduceContinue();
+                _send100Continue = false;
+            }
+        }
+
+        protected ReadableBufferAwaitable ReadRequestBodyPipeAsync()
+        {
+            var awaitable = _context.RequestBodyPipe.Reader.ReadAsync();
+
+            if (!awaitable.IsCompleted)
```

Comment:
@anurse Ideally we would only check this once per-request like we do for HTTP/1.1 instead of per-read like is done here. This might not be so easy with multiple streams per connection. I'll look into merging this tomorrow.

Reply:
You mean checking `_send100Continue` I assume?